lib/widgets/inbox.dart

```diff
@@ -377,17 +380,44 @@ class _DmItem extends StatelessWidget {
   }
 }
 
-class _StreamHeaderItem extends _HeaderItem {
+class StreamHeaderItem extends _HeaderItem {
   final Subscription subscription;
 
-  const _StreamHeaderItem({
+  /// This context is used to ensure the [_StreamSection] that encloses
+  /// the current [StreamHeaderItem] is visible after being collpased
+  /// through this [StreamHeaderItem] being used as a sticky header.
+  ///
+  /// A non-null value should be passed if this [StreamHeaderItem]
+  /// is used as a sticky header.
+  final BuildContext? parentContext;
+
+  const StreamHeaderItem({
     required this.subscription,
     required super.collapsed,
     required super.pageState,
```

Comment:
The preceding change has led to showing a lint warning for line no. 397 saying: `Invalid use of a private type in a public API.
Try making the private type public, or making the API that uses the private type also be private`.
I haven't touched it anymore so to get your feedback about it.